pinot-segment-local/src/main/java/org/apache/pinot/segment/local/io/util/VarLengthValueWriter.java

```diff
@@ -115,6 +119,9 @@ public void add(byte[] value)
 
   public void add(byte[] value, int length)
       throws IOException {
+    _totalLength += length;
+    Preconditions.checkArgument(_totalLength < Integer.MAX_VALUE,
```

Comment:
We can probably simply use the current position of the buffer to check if it will overflow